src/adapter/src/catalog.rs

```diff
@@ -4433,6 +4433,16 @@ impl<S: Append> Catalog<S> {
     pub fn system_config(&self) -> &SystemVars {
         self.state.system_config()
     }
+
+    pub async fn most_recent_storage_usage_collection(&self) -> Result<Option<EpochMillis>, Error> {
+        Ok(self
+            .storage()
+            .await
+            .storage_usage()
+            .await?
+            .map(|usage| usage.timestamp())
+            .max())
```

Comment:
The performance of this is a bit scary, since it'll get worse over time. It's not clear right now how to use a stash api to have the filtering be done by cockroach (so it returns the single row), but we might need to do this in the future. If we're at 1 hour collection and one entry per source, that's 720 entries per month per source. With 10 sources that's 86k entries per year, all of which would be queried each hour on all deployments. Clearly not a big deal now, but I have a light guess this will start to bite us in the future.

I don't have any suggestions for now since this use case doesn't fit very well into the current stash abstraction.

Reply:
I have an idea. We really only need to query the stash on startup. For all the other uses, it will be immediately after inserting the most recent row. So we can use the call-back in `catalog_transact` to have it return the timestamp directly from the transaction which should be faster. I'll try that out.

Reply:
Actually, shouldn't it be fine to just grab this value from the in-memory catalog? That would be much faster.

Reply:
We're only meant to be storing 60 days of data here! Adding that garbage collection is TODO. But that's how I'd recommend solving this. Lets us kick the can down the road on making improvements to stash.

Reply:
Also worth noting that the 1 hour collection interval is a nice-to-have -- I believe our only hard requirement right now is a collection once every 24 hours.